lib/Chat/ChatManager.php

```diff
@@ -154,11 +153,8 @@ public function sendMessage(Room $chat, Participant $participant, string $actorT
 			$comment->setParentId($replyTo->getId());
 		}
 
-		$this->dispatcher->dispatch(self::class . '::preSendMessage', new GenericEvent($chat, [
-			'comment' => $comment,
-			'room' => $chat,
-			'participant' => $participant,
-		]));
+		$event = new ChatParticipantEvent($chat, $comment, $participant);
+		$this->dispatcher->dispatch(self::class . '::preSendMessage', $event);
```

Comment:
To prevent typos and make refactoring easier you could add a static method to the event classes and use it like

```php
$this->dispatcher->dispatch(ChatParticipantEvent::preSendMessageEventName(), $event);
```

(function instead of constant so you can use ``self::class``)

Reply:
Oh you can actually use a constant. `public const EVENT_NAME = self::class . '...';` works just fine :)